src/login/logind-session.c

```diff
@@ -798,17 +798,17 @@ int session_start(Session *s, sd_bus_message *properties, sd_bus_error *error) {
         user_elect_display(s->user);
 
         /* Save data */
-        session_save(s);
-        user_save(s->user);
+        (void) session_save(s);
+        (void) user_save(s->user);
```

Comment:
Hmm, we always ignore return values of `*_save`. I vaguely recall that we don't need to cast to void in such case?

Reply:
so, the rules are a bit fuzzy, but basically if something can return errors but which we knowingly ignore then we should cast to (void) explicitly to tell the static analyzer about this. 

There are some (rare!) exceptions to this rule. log_error() and friends do return an error code, but it would be nasty to always cast it away, so we don't there. we have *so* many calls of that after all.

I would say, only those functions qualify for these exceptions which anyone touching the systemd codebase must have came into contact with already. Log functions are like that, they are *everywhere*. But this code here is specific to logind, only those folks which looked at logind specifically will know what the error return situation on session_save() and friends is. And thus I think we should use the verbose form here, and indicate to the reader: yes, there's an error here, and yes we knowingly ignore it.

Does that make any sense?